flang/lib/Lower/IntrinsicCall.cpp

```diff
@@ -2838,6 +2843,37 @@ mlir::Value IntrinsicLibrary::genSign(mlir::Type resultType,
   return builder.create<mlir::SelectOp>(loc, cmp, neg, abs);
 }
 
+// SIZE
+fir::ExtendedValue
+IntrinsicLibrary::genSize(mlir::Type resultType,
+                          llvm::ArrayRef<fir::ExtendedValue> args) {
+  // TODO: handle assumed-rank arrays, especially a dummy whose actual argument
+  // is an assumed-size array
```

Comment:
`fir::SequenceType` has a member function `hasUnknownShape()` to test if it is modeling an assumed rank array.

Reply:
Thanks!  I'll take a look.

Reply:
Actually, it looks like all dummy arguments that are assumed-rank get caught in CallInterface.cpp at around line 695 in "handleExplicitDummy()".  So there's no way for me to test code that makes a similar check in "genSize()".